### ISSUE TITLE
Handle case where account hasn't been created

### ISSUE THREAD

**Issue by msfeldstein** [stellar-deprecated/sep24-demo-client#95]
It's possible to create a deposit on an account that hasn't been created yet.  In that case, we depend on the anchor to create the account, which we need to wait for, then add the trustline, then complete the rest of the deposit. 

@accordeiro explains here: https://github.com/stellar/sep6-demo-client/pull/93
```
The deposit flow does not require a Stellar account to be created upfront, i.e. you can provide the public key of an account that does not exist until that moment (similarly to what we do with Friendbot). In this case, it'll be the anchor's responsibility to create that Stellar account and provide it with the minimum recommended funding (2.01XLM).

So I think the correct flow would be:

    Check /info
    Get /deposit (using the key provided by the config, regardless of it being an existing account)
    Interactive deposit flow
    Start polling transactions
    If we get a pending_trust as a response, then we actually perform the trustline creation

Does that make sense? The code is a bit messy, but I think this describes the exact moment in which we should establish a trustline: 

```

**Comment by accordeiro** [comment]
Thanks @msfeldstein! 

I wonder if we should increase the priority of this change, since on testnet it's easy to get some initial funding for an account (through Friendbot + lab) but on mainnet we'll probably see more cases where wallets expect the account creation / funding to be performed by that anchor – as otherwise they'd need to buy real XLM through some other means. 

Considering that, we probably need to ensure new anchors are properly implementing this account creation and initial funding flow. Does that make sense? @tomerweller @tomquisel I'd love to hear your thoughts as well.

**Comment by msfeldstein** [comment]
I think this is probably the next highest priority thing

**Comment by tomquisel** [comment]
I agree, this is high priority. It shouldn't be much of an issue for our mobile wallet since it'll pre-fund accounts, but it'll be a real stumbling block with any other wallet.
